src/core/network.js

```diff
@@ -115,20 +130,22 @@ var NetworkManager = (function NetworkManagerClosure() {
         pendingRequest.expectedStatus = 200;
       }
 
-      if (args.onProgressiveData) {
-        // Some legacy browsers might throw an exception.
-        try {
-          xhr.responseType = 'moz-chunked-arraybuffer';
-        } catch(e) {}
-        if (xhr.responseType === 'moz-chunked-arraybuffer') {
-          pendingRequest.onProgressiveData = args.onProgressiveData;
-          pendingRequest.mozChunked = true;
-        } else {
-          xhr.responseType = 'arraybuffer';
-        }
+//#if GENERIC
+      if (args.onProgressiveData && supportsMozChunked) {
+//#endif
+//#if (FIREFOX || B2G || MOZCENTRAL)
+//    if (args.onProgressiveData) {
+//#endif
+//#if (GENERIC || FIREFOX || B2G || MOZCENTRAL)
```

Comment:
This is *obviously* a matter of personal preference, but in my opinion all these preprocessor tags slightly hurt readability of this code block. Hence I think I'd prefer doing what I suggested in https://github.com/mozilla/pdf.js/pull/5531#discussion_r31856800 instead, since this (again, in my opinion) would significantly simplify the current code block.

Also, since the `B2G` preprocessor tag is being removed in PR #5947, I don't think you need to bother adding it here.